WordPress/wordpress-develop
# [53407] Widgets editor: decode HTML entities for name and description props

Trac ticket: https://core.trac.wordpress.org/ticket/53407

## Description

HTML entities appear in Legacy Widget's description. For example: 

`A monthly archive of your site&#8217;s Posts.` 😱 

This commit converts widget name and description HTML entities to their corresponding characters in the REST API response using [html_entity_decode](https://www.php.net/manual/en/function.html-entity-decode.php).

The decoding occurs before the call to `prepare_item_for_response` which sanitizes the value.

### Before

<img width="324" alt="Screen Shot 2021-06-15 at 3 56 34 pm" src="https://user-images.githubusercontent.com/6458278/122031674-435f8a00-ce12-11eb-9034-66bdb2b3eadb.png">

### After
<img width="344" alt="Screen Shot 2021-06-15 at 3 57 04 pm" src="https://user-images.githubusercontent.com/6458278/122031708-48bcd480-ce12-11eb-9434-f294c1359f51.png">

## Testing

1. Head over to `/wp-admin/themes.php?page=gutenberg-widgets` and select a widget block.
2. Check out the description in the inspector controls.

```
npm run test:php tests/phpunit/tests/rest-api/rest-widget-types-controller.php
```



---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**
